### PR TITLE
Feature/upgrade dependencies

### PR DESCRIPTION
**Pull request by hugomrazek**
Upgraded library dependencies, upgraded whole project to React 18.

Note that app has visual bugs (for example some forms have overlapping input labels and values due to wrongly passed default values) that will be resolved in upcoming refactor.

### CODE REVIEW

**Comment by blcham** [comment]
@kostobog have a look as well :)

**Comment by kostobog** [comment]
LGTM @blcham @hugomrazek

File: public/index.html [hunk 1]
```diff
@@ -10,10 +10,10 @@
     <link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/jointjs/3.2.0/joint.css" />
     <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width"/>
     <!-- Runtime environment variables -->
-    <script src="./config.js"></script>
+    <script src="/config.js"></script>
```
Comment (blcham) [inline public/index.html:13]: not sure if this is good idea ! This assumes that everything is in root, but what if we deploy it to e.g.
https://kbss.felk.cvut.cz/19msmt-demo/fta-fmea

File: public/index.html [hunk 2]
```diff
@@ -10,10 +10,10 @@
...
 </head>
 <body>
 <div id="root"></div>
-<script src="./build/bundle.js" type="text/javascript"></script>
+<script src="/build/bundle.js" type="text/javascript"></script>
```
Comment (blcham) [inline public/index.html:17]: same as above

**Review by blcham — COMMENTED** [review]
LGTM, just see mu comments

**Comment by kostobog** [comment]
@blcham 
@chekaval 

I tested this PR using deploy/docker-compose. The docker image did not work.
I added some fixes that I had locally to the PR.
The current version of the PR works. Tested in deploy/docker-compose with fta-fmea server versions :
- ghcr.io/kbss-cvut/fta-fmea:latest
- ghcr.io/kbss-cvut/fta-fmea:pr-48 ([Upgrade to jopa 1.1.4 and java 17](/kbss-cvut/fta-fmea/pull/48))

**Comment by blcham** [comment]
@kostobog 

Did you mean:
ghcr.io/kbss-cvut/fta-fmea-ui:pr-121 instead of ghcr.io/kbss-cvut/fta-fmea:latest  ?

**Comment by kostobog** [comment]
No I did not mean that. However, I updated the comment so it makes more sense.

**Comment by kostobog** [comment]
Also fixes #125

**Pull request merged**